Utilities: Some modernisation

AtkinsSJ opened the pull request:
Some DeprecatedString migration and a couple of other minor things.

(Previously also: removing a bunch of libc includes.)

krkk commented on Userland/Utilities/cksum.cpp:
```diff
@@ -4,26 +4,26 @@
  * SPDX-License-Identifier: BSD-2-Clause
```
there's a typo in “cksum: Stop using DeprecatedString”:
> “**COn** the other hand[…]”

trflynn89 commented on Userland/Utilities/stat.cpp:
```diff
@@ -10,10 +10,7 @@
 #include <LibCore/DateTime.h>
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
-#include <grp.h>
-#include <pwd.h>
```
Same here, both of these are directly used, for example.

AtkinsSJ commented:
Finally got around to modifying this. (Thanks for the update stalebot!)

- Dropped the "remove libc includes" commit.
- Fixed the typo krkk noticed.

AtkinsSJ commented:
Rebased and fixed conflicts.
